### PR TITLE
Update Template test

### PR DESCRIPTION
**Pull request by andyduong1920**
## What happened
We need to adjust the template test a bit in order to create more addons (web addons, API addons,...) in the next coming PR, also separate the job to Variant test and Template test, it's easy to understand than the current version

 
## Insight
- Update the README content, so contribute will know where to put the spec file.
- Adjust the Makefile to test addons and variant
 

## Proof Of Work
1/ The test should be passed.
image

2/ Readme content

Screen Shot 2020-10-08 at 17 14 42

### CODE REVIEW

**Review by rosle — APPROVED** [review]

**Review by olivierobert — COMMENTED** [review]
Looks good

File: Makefile [hunk 1]
```diff
@@ -6,21 +6,28 @@ build:
...
 	docker-compose -f docker-compose.test.yml up --detach db redis && \
 	docker-compose -f docker-compose.test.yml run --detach test bin/start.sh && \
 	cd ../.template && \
 	bundle install; \
 	if [ $(VARIANT) = web ]; then \
-	  bundle exec rspec --pattern="$(base_spec),spec/variants/web/**/*_spec.rb"; \
+		bundle exec rspec --pattern="$(base_addon_spec),${base_spec},$(web_addon_spec),$(web_spec)"; \
 	elif [ $(VARIANT) = api ]; then \
-	  bundle exec rspec --pattern="$(base_spec),spec/variants/api/**/*_spec.rb"; \
+		bundle exec rspec --pattern="$(base_addon_spec),${base_spec},$(api_addon_spec),$(api_spec)"; \
```
Comment (olivierobert) [inline Makefile:30]: Minor bit but the ordering of the spec files (which does not matter to be honest) is a bit counterintuitive in the sense that I would place base_spec first, then variant spec, then base_addon_spec and finally variant_addon_spec (from base to specific variant):

```
bundle exec rspec --pattern="${base_spec}, ${api_spec}, ${base_addon_spec}, $(api_addon_spec)";

bundle exec rspec --pattern="${base_spec}, ${web_spec}, ${base_addon_spec}, $(web_addon_spec)";
```

File: Makefile [hunk 2]
```diff
@@ -6,21 +6,28 @@ build:
...
+
+test_template:
 	cd $(APP_NAME) && \
 	docker-compose -f docker-compose.test.yml up --detach db redis && \
 	docker-compose -f docker-compose.test.yml run --detach test bin/start.sh && \
 	cd ../.template && \
 	bundle install; \
 	if [ $(VARIANT) = web ]; then \
-	  bundle exec rspec --pattern="$(base_spec),spec/variants/web/**/*_spec.rb"; \
+		bundle exec rspec --pattern="$(base_addon_spec),${base_spec},$(web_addon_spec),$(web_spec)"; \
```
Comment (olivierobert) [inline Makefile:28]: Is it intended to have a mix of `()` and `{}` 🤔 Otr should it be:

```suggestion
		bundle exec rspec --pattern="${base_addon_spec},${base_spec},${web_addon_spec},${web_spec}"; \
```

**Review by olivierobert — COMMENTED** [review]
Looks good overall. Just some minor comments.

File: Makefile (hunk 2 above)
Comment (andyduong1920) [reply Makefile:28]: Oh my bad, will update it :)

**Review by olivierobert — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (andyduong1920) [reply Makefile]: Fixed in https://github.com/nimblehq/rails-templates/pull/248/commits/c87b51926b9f1d643ccb7473b4970c56dbf26aa3 @olivierobert

File: Makefile (hunk 1 above)
Comment (andyduong1920) [reply Makefile]: Fixed in https://github.com/nimblehq/rails-templates/pull/248/commits/c87b51926b9f1d643ccb7473b4970c56dbf26aa3 @olivierobert

**Comment by olivierobert** [comment]
@andyduong1920 It would be great to port this changes to the Github actions PR once merged 🙏

**Comment by andyduong1920** [comment]
> @andyduong1920 It would be great to port this changes to the Github actions PR once merged 🙏

I will forward it to Hoang as he maintains that PR :D

**Pull request merged**